nil/services/rpc/jsonrpc/txpool_api_test.go

```diff
@@ -0,0 +1,106 @@
+package jsonrpc
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/NilFoundation/nil/nil/common/logging"
+	"github.com/NilFoundation/nil/nil/internal/db"
+	"github.com/NilFoundation/nil/nil/internal/types"
+	"github.com/NilFoundation/nil/nil/services/rpc/rawapi"
+	"github.com/NilFoundation/nil/nil/services/txnpool"
+	"github.com/stretchr/testify/suite"
+)
+
+type SuiteTxnPoolApi struct {
+	SuiteAccountsBase
+	txnpoolApi *TxPoolAPIImpl
+	api        *rawapi.NodeApiOverShardApis
+	pool       txnpool.Pool
+}
+
+const defaultMaxFee = 500
+
+func newTransaction(address types.Address, seqno types.Seqno, priorityFee uint64, code types.Code) *types.Transaction {
+	return &types.Transaction{
+		TransactionDigest: types.TransactionDigest{
+			To:                   address,
+			Seqno:                seqno,
+			MaxPriorityFeePerGas: types.NewValueFromUint64(priorityFee),
+			MaxFeePerGas:         types.NewValueFromUint64(defaultMaxFee),
+			Data:                 code,
+		},
+	}
+}
+
+func (suite *SuiteTxnPoolApi) SetupSuite() {
+	suite.SuiteAccountsBase.SetupSuite()
+	var err error
+	ctx := context.Background()
+
+	suite.pool, err = txnpool.New(ctx, txnpool.NewConfig(types.MainShardId), nil)
+	suite.Require().NoError(err)
+
+	database, err := db.NewBadgerDbInMemory()
+	suite.Require().NoError(err)
+	defer database.Close()
+
+	localShardApis := map[types.ShardId]rawapi.ShardApi{
+		types.MainShardId: rawapi.NewLocalShardApi(types.MainShardId, database, suite.pool),
+	}
+
+	shardApis := make(map[types.ShardId]rawapi.ShardApi)
+	localShardApi := rawapi.NewLocalShardApi(types.MainShardId, database, suite.pool)
+
+	shardApis[types.MainShardId], err = rawapi.NewLocalRawApiAccessor(types.MainShardId, localShardApi)
+
+	localApi := rawapi.NewNodeApiOverShardApis(localShardApis)
+	suite.txnpoolApi = NewTxPoolAPI(localApi, logging.NewLogger("Test"))
+	suite.api = rawapi.NewNodeApiOverShardApis(shardApis)
+	suite.Require().NoError(err)
+}
+
+func (suite *SuiteTxnPoolApi) TearDownSuite() {
+	suite.SuiteAccountsBase.TearDownSuite()
+}
+
+func (suite *SuiteTxnPoolApi) TestTnxApi() {
+	ctx := context.Background()
+	transactionAmount := uint64(10)
+
+	for i := range transactionAmount {
+		addr := types.ShardAndHexToAddress(0, "deadbeef"+fmt.Sprintf("%02d", i))
+		txn := newTransaction(addr, 0, 123, types.Code{byte(i)})
+		_, err := suite.pool.Add(ctx, txn)
+		suite.Require().NoError(err)
+	}
+
+	suite.Run("NodeApi", func() {
```

Comment:
Am I correct in understanding that this is a separate test on RawAPI so that our checks are more granular than when checking only the JSON-RPC API?

Reply:
Yes! Now we also have different output format for them.